### ISSUE TITLE
Grom not giving unique faction ability when interrogated by Eltharion

### ISSUE THREAD

**Issue by thefedexman** [chadvandy/cbfm_wh3#256]
**Describe the bug**
When capturing and interrogating Grom, you do not receive his unique mistwalker ability as Eltharion. You receive nothing

**Expected behavior**
You should get a unique mistwalker trait that mirrors Grom's Waagh ability

**To Reproduce**
1. Play as Yvresse faction
2. capture grom in battle via the Warden's Cage
3. Interrogate Grom

**Screenshots**
Screenshots attached of Grom being interrogated and of campaign overview with no unique ability in the faction bar

**Saves**
This is very easy to reproduce

**Additional context**
no other context required
![20220910004649_1](https://user-images.githubusercontent.com/113228387/189471181-27e0d8c1-7bcd-4599-acd5-c2818c1e0ce9.jpg)
![20220910004700_1](https://user-images.githubusercontent.com/113228387/189471182-8b20b370-5ca2-4ef9-b61f-103a86837a5a.jpg)

**Comment by thefedexman** [comment]
I'm not sure if this helps, but in the wh2_dlc15_eltharion_lair.lua file, this is the string that looks like it's associated with that specific campaign bonus you get from interrogating grom:

elseif action:starts_with("wh2_dlc15_prison_action_interrogate") then
				local current_effect = lair_interrogated_effect;
				
				for i = 1, #lair_action_effects do
					if lair_action_effects[i] ~= current_effect and warden:has_effect_bundle(lair_action_effects[i]) then
						lair_interrogated_effect = lair_action_effects[i];
					end
				end
				
				lair_RemoveMistwalkerAbilities(warden, lair_interrogated_effect);
				lair_interrogated_prisoner = prisoner_cqi;
				
				if lair_improved_interrogate then
					local mf_list = warden:military_force_list();
					
					for i = 0, mf_list:num_items() - 1 do
						local current_mf = mf_list:item_at(i);
						
						if not current_mf:is_armed_citizenry() and current_mf:has_general() then
							cm:add_agent_experience(cm:char_lookup_str(current_mf:general_character()), 250);
						end
					end
				end
				
				**if is_grom then
					cm:apply_effect_bundle("wh2_dlc15_hef_dungeon_mistwalker_upgrade_greenskins_grom", yvresse_faction_key, 0);
				end**

**Comment by thefedexman** [comment]
this issue still persists as of patch 2.2

**Comment by LordOmlette** [comment]
The problem is immediately after the code snippet that thefedexman shared. In lair_UpdatePrisonerEffects, the reward for interrogating Grom is blindly removed without bothering to check whether he's one of the prisonerse.

**Comment by LordOmlette** [comment]
This worked for me:
```
local old_lair_UpdatePrisonerEffects = lair_UpdatePrisonerEffects;
lair_UpdatePrisonerEffects = function (faction)
	old_lair_UpdatePrisonerEffects (faction);

	for _, prisoner in model_pairs (cm:model():prison_system():get_faction_prisoners (faction)) do
		local cc = prisoner:character();
		if cc:character_subtype ("wh2_dlc15_grn_grom_the_paunch") and not faction:has_effect_bundle ("wh2_dlc15_hef_dungeon_mistwalker_upgrade_greenskins_grom") then
			cm:apply_effect_bundle ("wh2_dlc15_hef_dungeon_mistwalker_upgrade_greenskins_grom", "wh2_main_hef_yvresse", 0);
			break;
		end
	end
end	

```

**Comment by thefedexman** [comment]
oh hell yea. you guys are honestly great and I appreciate all the hard work you do fixing the ridiculous amount of bugs in this game

**Comment by mfvreeland** [comment]
@LordOmlette 
Did you submit this as a pull request at any point, or just fix it on your end as a test? If it's the latter, do you mind if I implement into cbfm?

**Comment by Harmandodelavega** [comment]
not fixed in 2.2, 2.3 tbd

**Comment by thefedexman** [comment]
I've been testing this every update, and I'm about to test it right now. I'll update the thread to save you all some time

**Comment by thefedexman** [comment]
yea it's still not fixed. I realized I could have probably just done a packfile compare, but I tested it in-game
![20221122143917_1](https://user-images.githubusercontent.com/113228387/203416860-bb1306e8-3889-49dd-83fb-46d7bd0d239c.jpg)

**Comment by mfvreeland** [comment]
@thefedexman Do you have a save I could use to test this? I'd like to implement the script fix, but I need to test it to make sure it works before we can add the fix to main.

**Comment by thefedexman** [comment]
@mfvreeland I'll create one tonight for you. Can I attach it to the thread?

**Comment by mfvreeland** [comment]
> @mfvreeland I'll create one tonight for you. Can I attach it to the thread?

Yes, although you will need to put it into a .ZIP file first, since GitHub won't take a regular save file. Alternatively, you can post directly on Discord.

**Comment by thefedexman** [comment]
sorry guys I got tied up with some other stuff. I'm getting a save for you right now. I'll do two; one pre-grom battle and one post when he's in eltharion's custody

**Comment by thefedexman** [comment]
[New folder.zip](https://github.com/chadvandy/cbfm_wh3/files/10107989/New.folder.zip)

there are two save files in here, labeled pre and post. I was using mods with these. screenshot below with the mods being used. recommend warden's cage mod if you want to autoresolve to capture grom and not fight the battle out

![image](https://user-images.githubusercontent.com/113228387/204394702-9964daab-c266-4994-8f2b-f98330ce2856.png)

**Comment by LordOmlette** [comment]
Using the save files @thefedexman provided, I can confirm this is fixed in 2.4.0.

![image](https://user-images.githubusercontent.com/112673788/219789050-439fa614-a1b4-411b-8356-918f29abeeef.png)

**Comment by mfvreeland** [comment]
According to the notes, this should be fixed in 2.4. Needs confirmation.

**Comment by mfvreeland** [comment]
Confirmed fixed in 2.4.
